apollo-router/src/plugins/coprocessor/supergraph.rs

```diff
@@ -0,0 +1,881 @@
+                    .and_id(TraceId::maybe_new().map(|id| id.to_string()))
+                    .and_body(body_to_send)
+                    .and_context(context_to_send)
+                    .and_sdl(generator_sdl_to_send)
+                    .build();
+
+                // Second, call our co-processor and get a reply.
+                tracing::debug!(?payload, "externalized output");
+                let guard = generator_map_context.enter_active_request();
+                let co_processor_result = payload
+                    .call(generator_client, &generator_coprocessor_url)
+                    .await;
+                drop(guard);
+                tracing::debug!(?co_processor_result, "co-processor returned");
+                let co_processor_output = co_processor_result?;
+
+                validate_coprocessor_output(
+                    &co_processor_output,
+                    PipelineStep::SupergraphResponse,
+                )?;
+
+                // Third, process our reply and act on the contents. Our processing logic is
+                // that we replace "bits" of our incoming response with the updated bits if they
+                // are present in our co_processor_output. If they aren't present, just use the
+                // bits that we sent to the co_processor.
+                let new_deferred_response: crate::response::Response =
+                    match co_processor_output.body {
+                        Some(value) => serde_json::from_value(value)?,
+                        None => deferred_response,
+                    };
+
+                if let Some(context) = co_processor_output.context {
+                    for (key, value) in context.try_into_iter()? {
+                        generator_map_context.upsert_json_value(key, move |_current| value);
+                    }
+                }
+
+                // We return the deferred_response into our stream of response chunks
+                Ok(new_deferred_response)
+            }
+        })
+        .map(|res: Result<response::Response, BoxError>| match res {
+            Ok(response) => response,
+            Err(e) => {
+                tracing::error!("coprocessor error handling deferred supergraph response: {e}");
+                response::Response::builder()
+                    .error(
+                        Error::builder()
+                            .message("Internal error handling deferred response")
+                            .extension_code("INTERNAL_ERROR")
+                            .build(),
+                    )
+                    .build()
+            }
+        });
+
+    //response.response = http::Response::from_parts(parts, stream::once(future::ready(new_body)));
+
+    // Create our response stream which consists of the bytes from our first body chained with the
+    // rest of the responses in our mapped stream.
+    //let bytes = hyper::body::to_bytes(body).await.map_err(BoxError::from);
```

Comment:
can this be removed?

Reply:
fixed in [2680e59](https://github.com/apollographql/router/pull/3408/commits/2680e5933309b59c47753b58eb1c9b4b80d254f1)